nilearn/plotting/surf_plotting.py

```diff
@@ -1362,7 +1380,7 @@ def plot_img_on_surf(stat_map, surf_mesh='fsaverage5', mask_img=None,
 
     Parameters
     ----------
-    stat_map : str or 3D Niimg-like object
+    stat_map : :obj:`str` or :class:`pathlib.Path` or 3D Niimg-like object
```

Comment:
Tested that the code works with both string and path so updated the doc.

Reply:
On that note, it may be worth checking in the public API that users can also pass a Path in places where they can pass a string to a file.

I have not checked but it seems that many functions require a string and that Paths are not supported (or at least the doc string is not necessarily explicit - or I missed it).

May be done as some internal refactoring to rely more on pathlib rather than os (where relevant) - to be discussed.

Reply:
Yes, we should open an issue and address that later.